app/services/cloud_front_header_parser.rb

```diff
@@ -0,0 +1,34 @@
+class CloudFrontHeaderParser
+  def initialize(request)
+    @request = request
+  end
+
+  def client_ip
+    addr = ip_and_port
+    return nil unless addr
+    addr[:ip]
+  end
+
+  def client_port
+    addr = ip_and_port
+    return nil unless addr
+    addr[:port]
+  end
```

Comment:
`client_ip` is not used, either right? I'd just remove it entirely and even remove `#ip_and_port` and just throw out the client IP address for now

```suggestion
  def client_ip
    addr = ip_and_port
    return nil unless addr
    addr[:ip]
  end

  def client_port
    *_parts, port = viewer_address&.split(':')
    port.presence
  end
```

Reply:
Do you know if anyone on the ops side would be able to peek at some of the data and see how it's being reported?

I'm _assuming_ that `request.client_ip` (also used elsewhere in the code) is going to parse X-Forwarded-For properly in a way that matches the "real" client IP from this header, but I've seen enough cowboy stuff done in load balancers and caches to not entirely trust it without someone confirming we can.

Reply:
https://github.com/18F/identity-devops/issues/5079 has some example header values that are logged, is that useful?